crest/Assets/Crest/Crest/Scripts/Collision/CollProvider.cs

```diff
@@ -33,7 +38,12 @@ public interface ICollProvider
         /// <param name="o_resultDisps">Displacement vectors for water surface points that will displace to the XZ coordinates of the query points. Water heights are given by sea level plus the y component of the displacement.</param>
         /// <param name="o_resultNorms">Water normals at the query positions. Pass null if this information is not required.</param>
         /// <param name="o_resultVels">Water surface velocities at the query positions. Pass null if this information is not required.</param>
+#if CREST_BURST_QUERY
+        /// <param name="useNormals"></param>
+        int Query(int i_ownerHash, float i_minSpatialLength, ref NativeArray<Vector3> i_queryPoints, ref NativeArray<Vector3> o_resultDisps, ref NativeArray<Vector3> o_resultNorms, ref NativeArray<Vector3> o_resultVels, bool useNormals);
```

Comment:
"useNormals" should be dropped or also added to the other method to keep the API consistent. I would vote with dropping unless it really saves on performance.

Reply:
That being said, checking null or Length to determine intent of the user throughout the call stack makes things very confusing down the line.

Reply:
Went with dropping it and improved checks throughout. Also added a helper to make passing "null" easier.